subsys/bluetooth/controller/ll_sw/ull_conn_iso.c

```diff
@@ -324,3 +394,155 @@ void ull_conn_iso_resume_ticker_start(struct lll_event *resume_event,
 	LL_ASSERT((ret == TICKER_STATUS_SUCCESS) ||
 		  (ret == TICKER_STATUS_BUSY));
 }
+
+static void disabled_cig_cb(void *param)
+{
+	struct ll_conn_iso_group *cig;
+
+	cig = param;
+
+	ll_conn_iso_group_release(cig);
+
+	/* TODO: Flush pending TX in LLL */
+}
+
+static void ticker_stop_op_cb(uint32_t status, void *param)
+{
+	static memq_link_t link;
+	static struct mayfly mfy = {0, 0, &link, NULL, NULL};
+	struct ll_conn_iso_group *cig;
+	struct ull_hdr *hdr;
+	uint32_t ret;
+
+	/* Assert if race between thread and ULL */
+	LL_ASSERT(status == TICKER_STATUS_SUCCESS);
+
+	cig = param;
+	hdr = &cig->ull;
+	mfy.param = cig;
+
+	if (ull_ref_get(hdr)) {
+		/* Event active (prepare/done ongoing) - wait for done and
+		 * disable there. Abort the ongoing event in LLL.
+		 */
+		LL_ASSERT(!hdr->disabled_cb);
+		hdr->disabled_param = mfy.param;
+		hdr->disabled_cb = disabled_cig_cb;
+
+		mfy.fp = lll_disable;
+		ret = mayfly_enqueue(TICKER_USER_ID_ULL_LOW,
+				     TICKER_USER_ID_LLL, 0, &mfy);
+		LL_ASSERT(!ret);
+	} else {
+		/* Disable now */
+		mfy.fp = disabled_cig_cb;
+		ret = mayfly_enqueue(TICKER_USER_ID_ULL_LOW,
+				     TICKER_USER_ID_ULL_HIGH, 0, &mfy);
+		LL_ASSERT(!ret);
+	}
+}
+
+static void disabled_cis_cb(void *param)
+{
+	struct ll_conn_iso_group *cig;
+	struct ll_conn_iso_stream *cis;
+	uint32_t ticker_status;
+	uint16_t handle_iter;
+	uint8_t cis_idx;
+	uint8_t num_cis;
+
+	cig = param;
+	num_cis = cig->lll.num_cis;
+	handle_iter = UINT16_MAX;
+
+	/* Remove all CISes marked for teardown */
+	for (cis_idx = 0; cis_idx < cig->lll.num_cis; cis_idx++) {
+		cis = ll_conn_iso_stream_get_by_group(cig, &handle_iter);
+		LL_ASSERT(cis);
+
+		if (cis->teardown) {
+			struct ll_conn *conn;
+			ll_iso_stream_released_cb_t cis_released_cb;
+
+			conn = ll_conn_get(cis->lll.acl_handle);
+			cis_released_cb = cis->released_cb;
+
+			ll_conn_iso_stream_release(cis);
+			cig->lll.num_cis--;
+
+			/* Check if removed CIS had an ACL disassociation callback. Invoke
+			 * the callback to allow cleanup.
+			 */
+			if (cis_released_cb) {
+				/* No more CISes associated with ACL - notify caller */
```

Comment:
It is not clear to me why there are no more CISes associated with the ACL at this point. It seems like this would be reached for every CIS that is torn down, but I may be overlooking something.

Reply:
No, you're rigth. The comment is from my first implementation where the callback wasn't invoked until last disassociation. Will change comment.